test/integration/TaikoDaoFactory.t.sol

```diff
@@ -0,0 +1,822 @@
+            pluginRepoFactory: PluginRepoFactory(address(pRefoFactory)), // PluginRepoFactory
+            // Plugin setup's
+            multisigPluginSetup: multisigPluginSetup,
+            emergencyMultisigPluginSetup: emergencyMultisigPluginSetup,
+            optimisticTokenVotingPluginSetup: optimisticTokenVotingPluginSetup,
+            // Multisig
+            multisigMembers: multisigMembers, // address[]
+            // ENS
+            stdMultisigEnsDomain: "multisig", // string
+            emergencyMultisigEnsDomain: "eMultisig", // string
+            optimisticTokenVotingEnsDomain: "optimistic" // string
+        });
+
+        // Deploy
+        TaikoDaoFactory factory = new TaikoDaoFactory(creationSettings);
+
+        factory.deployOnce();
+        TaikoDaoFactory.Deployment memory deployment = factory.getDeployment();
+
+        vm.roll(block.number + 1); // mint one block
+
+        // DAO checks
+
+        assertNotEq(address(deployment.dao), address(0), "Empty DAO field");
+        assertEq(deployment.dao.daoURI(), "", "DAO URI should be empty");
+        assertEq(address(deployment.dao.signatureValidator()), address(0), "signatureValidator should be empty");
+        assertEq(address(deployment.dao.getTrustedForwarder()), address(0), "trustedForwarder should be empty");
+        assertEq(
+            deployment.dao.hasPermission(
+                address(deployment.dao), address(deployment.dao), deployment.dao.ROOT_PERMISSION_ID(), bytes("")
+            ),
+            true,
+            "The DAO should be ROOT on itself"
+        );
+        assertEq(
+            deployment.dao.hasPermission(
+                address(deployment.dao), address(deployment.dao), deployment.dao.UPGRADE_DAO_PERMISSION_ID(), bytes("")
+            ),
+            true,
+            "The DAO should have UPGRADE_DAO_PERMISSION on itself"
+        );
+        assertEq(
+            deployment.dao.hasPermission(
+                address(deployment.dao),
+                address(deployment.dao),
+                deployment.dao.REGISTER_STANDARD_CALLBACK_PERMISSION_ID(),
+                bytes("")
+            ),
+            true,
+            "The DAO should have REGISTER_STANDARD_CALLBACK_PERMISSION_ID on itself"
+        );
+
+        // Multisig plugin
+
+        assertNotEq(address(deployment.multisigPlugin), address(0), "Empty multisig field");
+        assertEq(
+            deployment.multisigPlugin.lastMultisigSettingsChange(),
+            block.number - 1,
+            "Invalid lastMultisigSettingsChange"
+        );
+        assertEq(deployment.multisigPlugin.proposalCount(), 0, "Invalid proposal count");
+        assertEq(deployment.multisigPlugin.addresslistLength(), 13, "Invalid addresslistLength");
+        for (uint256 i = 0; i < 13; i++) {
+            assertEq(deployment.multisigPlugin.isMember(multisigMembers[i]), true, "Should be a member");
+        }
+        for (uint256 i = 14; i < 50; i++) {
```

Comment:
This one is tricky, but good to point this out.

The address at index `0` is actually `1`, to avoid granting permissions to `0x0`. Therefore, the 12th iteration of `multisigMembers` would contain `address(13)`, while the initial iteration (14) of the second loop does `address(uint160(i))`, which is 14

Reply:
Gotcha thanks for clarifying